### PR TITLE
Create usable workflows

### PR DESCRIPTION
**Pull request by christophfroehlich**
This PR implements a reusable workflow as proposed in https://github.com/ros-controls/ros2_controllers/pull/664#issuecomment-1587601522 

This should enable CI workflows in other repositories, which can check if the docs are building fine.

As an example, this works already with https://github.com/ros-controls/ros2_control_demos/pull/303

One has to explicitly state the repo in the reusable workflow, otherwise the job from the caller workflow looks inside its repo for the action/reusable workflow. (or have I missed someting?)
Because the workflow is not in the master branch, I had to use my fork -> I'll change that before the merge.

CI for this PR fails, because the Cleanup jobs switch to the master branch, where the composable action is not available obviously

### CODE REVIEW

File: .github/actions/install-dependencies/action.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: "Install Dependencies"
+description: "Python dependencies"
+
+runs:
+  using: 'composite'
+  steps:
+  - uses: actions/checkout@v3
+    with:
+      repository: christophfroehlich/control.ros.org
+      ref: reusable_wf
```
Comment (christophfroehlich) [inline .github/actions/install-dependencies/action.yml:10]:
```suggestion
      repository: ros-controls/control.ros.org
      ref: master
```

File: .github/workflows/reusable-sphinx-check-single-version.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        type: string
+
+jobs:
+  build-single-version:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        repository: christophfroehlich/control.ros.org
+        ref: reusable_wf
```
Comment (christophfroehlich) [inline .github/workflows/reusable-sphinx-check-single-version.yml:23]:
```suggestion
        repository: ros-controls/control.ros.org
        ref: master
```

File: .github/workflows/reusable-sphinx-check-single-version.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        fetch-depth: 0
+    - name: Install dependencies
+      uses: christophfroehlich/control.ros.org/.github/actions/install-dependencies@reusable_wf
```
Comment (christophfroehlich) [inline .github/workflows/reusable-sphinx-check-single-version.yml:26]:
```suggestion
      uses: ros-controls/control.ros.org/.github/actions/install-dependencies@reusable_wf
```

Comment (christophfroehlich) [inline .github/workflows/reusable-sphinx-check-single-version.yml:26]:
```suggestion
      uses: ros-controls/control.ros.org/.github/actions/install-dependencies@master
```

**Comment by bmagyar** [comment]
The CI on this should be green using your fork, right?

**Comment by christophfroehlich** [comment]
this one here won't go green unless it is merged into master. But in other repos, like in the linked one in the demos I think so yes

**Review by bmagyar — APPROVED** [review]

**Comment by bmagyar** [comment]
Let's see

**Comment by christophfroehlich** [comment]
@mergifyio backport humble

**Pull request merged**